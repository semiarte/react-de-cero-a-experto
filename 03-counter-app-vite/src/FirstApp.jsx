/* eslint-disable react/prop-types */
import PropTypes from 'prop-types';

const FirstApp = ({ title, subTitle, name }) => {

  // console.log(props);
  
  return (
    <>
      <h1 data-testid="test-title"> { title } </h1>
      {/* <code>{ JSON.stringify( newMessage ) }</code> */}
      <p>{ subTitle }</p>
      <p>{ subTitle }</p>
      <p>{ name }</p>
    </>
  )
}


FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
}

FirstApp.defaultProps = {
  name: 'Fernando Herrera',
  subTitle: 'No hay subtítulo',
  // title: 'No hay título',
}

export default FirstApp;