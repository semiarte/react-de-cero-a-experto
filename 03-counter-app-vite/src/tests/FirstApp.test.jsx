/* eslint-disable no-unused-vars */
import { render } from "@testing-library/react"
import FirstApp from "../FirstApp"

/* eslint-disable no-undef */
describe('Pruebas en <FirstApp />', () => { 
    test('debe de hacer match con el snapshot', () => { 
        
        const title = 'Hola, soy Goku'
        const container = render(<FirstApp title={ title } />)

        expect( container ).toMatchSnapshot();
     });

     test('debe mostrar el tÃ­tulo del h1', () => {

        const title = 'Hola soy Goku';
        const { container, getByText } = render(<FirstApp title={ title } />)
        expect ( getByText(title) ).toBeTruthy();

        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain( title );

     });

     test('debe de mostrar el subtitulo enviado por props', () => {
        
        const title = 'Hola, Soy Goku';
        const subTitle = 'Soy un subtitulo';
        const { getAllByText } = render( 
            <FirstApp 
                title={ title }
                subTitle={ subTitle }
            /> 
        );

        expect( getAllByText(subTitle).length ).toBe(2);

    });
 })