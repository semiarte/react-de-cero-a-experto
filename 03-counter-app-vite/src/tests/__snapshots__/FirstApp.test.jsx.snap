// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con el snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        data-testid="test-title"
      >
         
        Hola, soy Goku
         
      </h1>
      <p>
        No hay subtítulo
      </p>
      <p>
        No hay subtítulo
      </p>
      <p>
        Fernando Herrera
      </p>
    </div>
  </body>,
  "container": <div>
    <h1
      data-testid="test-title"
    >
       
      Hola, soy Goku
       
    </h1>
    <p>
      No hay subtítulo
    </p>
    <p>
      No hay subtítulo
    </p>
    <p>
      Fernando Herrera
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
